processing/src/main/java/io/druid/query/metadata/metadata/ListColumnIncluderator.java

```diff
@@ -64,12 +64,13 @@ public boolean include(String columnName)
     for (String column : this.columns) {
       final byte[] bytes = StringUtils.toUtf8(column);
       columns.add(bytes);
-      size += bytes.length;
+      size += bytes.length + 1;
     }
 
     final ByteBuffer bytes = ByteBuffer.allocate(size).put(LIST_CACHE_PREFIX);
     for (byte[] column : columns) {
       bytes.put(column);
+      bytes.put((byte) 0);
```

Comment:
Indeed, 0xFF and 0xFE can be used as separators, since they will never appear in a valid UTF-8 byte stream (those bytes are reserved to indicate byte order mark in UTF-16). Using a zero byte still make it technically possible to have cache collisions in case someone uses funky UTF-8 characters.